File: android/src/com/mygdx/game/voxel/reader/VoxelDataPerColor.java
package com.mygdx.game.voxel.reader;

/**
 * Created by daely on 11/23/2016.
 */
public class VoxelDataPerColor //struct in original C# code
{
    public int nvoxels;
    public byte[] x;
    public byte[] y;
    public byte[] z;
    public int voxelrgbacolor;
    public VoxelDataPerColor(int nvoxels,int rgbacolor) {
        this.nvoxels=0;
        x=new byte[nvoxels];
        y=new byte[nvoxels];
        z=new byte[nvoxels];
        this.voxelrgbacolor =rgbacolor;
    }

    public void addVoxel(byte xp, byte yp, byte zp) {
        int i=nvoxels++;
        x[i]=xp;
        y[i]=yp;
        z[i]=zp;
    }
}

File: android/src/com/mygdx/game/voxel/reader/VoxelDataPerColor.kt
package com.mygdx.game.voxel.reader

/**
 * Created by daely on 11/23/2016.
 * struct in original C# code
 */
class VoxelDataPerColor(nvoxels: Int, var voxelrgbacolor: Int) {
    var nvoxels: Int = 0
    var x: ByteArray
    var y: ByteArray
    var z: ByteArray

    init {
        this.nvoxels = 0
        x = ByteArray(nvoxels)
        y = ByteArray(nvoxels)
        z = ByteArray(nvoxels)
    }

    fun addVoxel(xp: Byte, yp: Byte, zp: Byte) {
        val i = nvoxels++
        x[i] = xp
        y[i] = yp
        z[i] = zp
    }
}
